core/src/ledger/ibc/context/common.rs

```diff
@@ -203,160 +482,139 @@ pub trait IbcCommonContext: IbcStorageContext {
 
-    /// Write the packet commitment
-    fn store_packet_commitment(
+    /// Store the packet acknowledgement
+    fn store_packet_ack(
         &mut self,
-        path: &CommitmentPath,
-        commitment: PacketCommitment,
-    ) -> Result<(), ContextError> {
-        let path = Path::Commitment(path.clone());
-        let key = storage::ibc_key(path.to_string())
-            .expect("Creating a key for the client state shouldn't fail");
-        let bytes = commitment.into_vec();
-        self.write(&key, bytes).map_err(|_| {
-            ContextError::PacketError(PacketError::Channel(
-                ChannelError::Other {
-                    description: format!(
-                        "Writing the packet commitment failed: Key {}",
-                        key
-                    ),
-                },
-            ))
-        })
+        port_id: &PortId,
+        channel_id: &ChannelId,
+        sequence: Sequence,
+        ack_commitment: AcknowledgementCommitment,
+    ) -> Result<()> {
+        let key = storage::ack_key(port_id, channel_id, sequence);
+        let bytes = ack_commitment.into_vec();
+        self.write_bytes(&key, bytes).map_err(ContextError::from)
     }
 
-    /// Write the NextSequenceSend
-    fn store_next_sequence_send(
+    /// Delete the packet acknowledgement
+    fn delete_packet_ack(
         &mut self,
-        path: &SeqSendPath,
-        seq: Sequence,
-    ) -> Result<(), ContextError> {
-        let path = Path::SeqSend(path.clone());
-        let key = storage::ibc_key(path.to_string())
-            .expect("Creating a key for the client state shouldn't fail");
-        self.store_sequence(&key, seq)
-    }
-
-    /// Increment and write the counter
-    fn increase_counter(&mut self, key: &Key) -> Result<(), ContextError> {
-        let count = self.read_counter(key)?;
-        self.write(key, (count + 1).to_be_bytes().to_vec())
-            .map_err(|_| {
-                ContextError::ClientError(ClientError::Other {
-                    description: format!(
-                        "Writing the counter failed: Key {}",
-                        key
-                    ),
-                })
-            })
+        port_id: &PortId,
+        channel_id: &ChannelId,
+        sequence: Sequence,
+    ) -> Result<()> {
+        let key = storage::ack_key(port_id, channel_id, sequence);
+        self.delete(&key).map_err(ContextError::from)
     }
 
-    /// Write the sequence
-    fn store_sequence(
-        &mut self,
-        key: &Key,
-        sequence: Sequence,
-    ) -> Result<(), ContextError> {
-        self.write(key, u64::from(sequence).to_be_bytes().to_vec())
-            .map_err(|_| {
-                ContextError::PacketError(PacketError::Channel(
-                    ChannelError::Other {
+    /// Read a counter
+    fn read_counter(&self, key: &Key) -> Result<u64> {
+        match self.read_bytes(key)? {
+            Some(value) => {
+                let value: [u8; 8] =
```

Comment:
this is odd. Why can't we just store the u64 directly?

Reply:
You're right. We can use u64 directly (I used it for sequences before. but it isn't used anymore.)